lib/vanity/experiment/ab_test.rb

```diff
@@ -617,13 +617,15 @@ def alternative_for(identity)
       # Saves the assignment of an alternative to a person and performs the
       # necessary housekeeping. Ignores repeat identities and filters using
       # Playground#request_filter.
-      def save_assignment(identity, index, request)
+      def save_assignment(identity, index, _request)
         return if index == connection.ab_showing(@id, identity)
+        return if connection.ab_seen @id, identity, index
```

Comment:
I spent some time looking at this, because this guard clause was lifted up out of the `#call_on_assignment_if_available` method. I think it's safe and an improvement because it ensures re-assigning the same assignment.